lib/matplotlib/image.py

```diff
@@ -671,6 +671,10 @@ def set_data(self, A):
             raise TypeError("Image data of dtype {} cannot be converted to "
                             "float".format(self._A.dtype))
 
+        if not (self._A.ndim == 3 and self._A.shape[-1]==1):
```

Comment:
```suggestion
        if (self._A.ndim == 3 and self._A.shape[-1]==1):
```

I think this logic is inverted?

Reply:
sorry! you are right!